makelib/xpkg.mk

```diff
@@ -64,8 +64,9 @@ define xpkg.build.targets
 xpkg.build.$(1):
 	@$(INFO) Building package $(1)-$(VERSION).xpkg for $(PLATFORM)
 	@mkdir -p $(OUTPUT_DIR)/xpkg/$(PLATFORM)
-	@$(UP) xpkg build \
-		--controller $(BUILD_REGISTRY)/$(1)-$(ARCH) \
+	@controller_arg=$$$$(grep -E '^kind:\s+Provider\s*$$$$' $(XPKG_DIR)/crossplane.yaml > /dev/null && echo "--controller $(BUILD_REGISTRY)/$(1)-$(ARCH)"); \
```

Comment:
I’m fine with doing this for now, but ultimately the fact that the kind is a Provider does not necessarily indicate that we want to specify a base controller image. The other option I was exploring was a separate list of values denoting which packages required a controller reference, which would be more work for the consumer of this package but also less opaque. Longer term I anticipate using a dedicated GitHub action for all of this machinery. Given the alternative solution, would you still prefer to go with solution provided here?

Reply:
yeah, I agree that this should not be the ultimate solution and think this as a stopgap solution. 

> the fact that the kind is a Provider does not necessarily indicate that we want to specify a base controller image

which is already the case, so this is PR is not changing something in that regard.

> Given the alternative solution, would you still prefer to go with solution provided here?

That was something I considered as well, but didn't want to build that since I didn't feel comfortable with keeping two lists. So wanted make a minor change to unblock myself without making things worse. 
So, yes. I would like to proceed if you agree with the above.

Reply:
That seems reasonable to me 👍🏻

Reply:
> Longer term I anticipate using a dedicated GitHub action for all of this machinery.

As a side note, I see some value with having in the build submodule compared to having it as a GitHub action like trying/testing things locally.